app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -345,6 +351,87 @@ protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
         return $magentoDerivedCartData;
     }
 
+    /**
+     * Checks if billing address of a quote has all the expected fields.
+     * If not, and therefore invalid, the address is replaced by using the
+     * provided shipping address.
+     *
+     * @param Mage_Sales_Model_Quote_Address $billingAddress   The billing address
+     *                                                         to be checked
+     * @param Mage_Sales_Model_Quote_Address $fallbackAddress  The address to fallback
+     *                                                         to in case of invalid billing
+     *                                                         address
+     *
+     *
+     * @return bool  true if a correction is made, otherwise false
+     */
+    public function correctBillingAddress(&$billingAddress, $fallbackAddress)
+    {
+        $quote = $billingAddress->getQuote();
+
+        $wasCorrected = false;
+
+        /////////////////////////////////////////////
+        /// Changes persisted to Magento and Bolt
+        /////////////////////////////////////////////
+        if ($fallbackAddress) {
+            if (
+                !trim($billingAddress->getStreetFull()) ||
```

Comment:
why these can be empty in the first place?

Reply:
@daisy1754 That's the million dollar question and I can answer it partially.  Initially, when any part of the billing was missing, the old code removed it entirely.  This is one way in which this use to happen.  We have fixed that problem.

A second way is when there is a store which is selling virtual products and we do not collect address information because of this.

A third way is if there are custom address entry forms or routines which for whatever reason, does not enforce billing address data and only is concerned with shipping data.  In Europe, for instance, it is common that billing address data is not associated with the credit card.

And finally, there maybe situations that currently I can't explain.  Indique is one of them.  It is quite possible that it was because of the first reason and Indique not updating their plugin, but because of the second two reasons listed, I don't see any harm in checking for this here.

What makes all this possible is that in the code, we turn off address validation for Bolt orders on order creation:

https://github.com/BoltApp/bolt-magento1/blob/develop/app/code/community/Bolt/Boltpay/Model/Order.php#L124-L125

This has been a long standing practice and I am not challenging that decision in this pull request, but it is a question to ask:  should we be conditionally validating these addresses?

Reply:
hmm, sounds like for 2 and 3, billing address is intended to be blank. are we certain billing address is still missing some field unintentionally with latest plugin?  I'd like to avoid code (which we need to maintain) if it's not needed.

Reply:
If you prefer to keep these, let's set bugsnag.

Reply:
@daisy1754 
Can we set it in bugsnag and schedule the code for removal if it is not triggered in bugsnag after a single quarter, let's say end of March?

Reply:
sounds good. let's create asana task to track code removal